.github/workflows/algolia/src/uploadToAlgolia.ts

```diff
@@ -1,9 +1,12 @@
-import { generateAlgoliaJson, uploadAlgoliaData } from ".";
+import { syncAlgoliaIndex } from "./syncAlgoliaIndex";
 
 const main = async () => {
-  const records = generateAlgoliaJson();
-  await uploadAlgoliaData(records);
-  console.log("Algolia Indexes Updated.");
+  try {
+    await syncAlgoliaIndex();
+    console.log("Algolia Indexes Updated.");
+  } catch (error) {
+    throw new Error(`Algolia Indexing Failed: ${error}`);
```

Comment:
I realized a failure state didn't exist. Added this now.

Reply:
Nice! Now that we have all the logic inside `syncAgoliaIndex`, how about moving `main` into it too? `syncAlgoliaIndex.ts` (or `sync-algolia-index.ts`) can become a self-sufficient CLI command that we can call from CI. It won’t need to export anything, just play all the music when someone calls `ts-node path/to/command.ts`.

If at some later point we decide to use `syncAlgoliaIndex` elsewhere, we’ll split the file into two without issues. Doing this prematurely does not help much because it’s impossible to guess how exactly we’ll want to reuse the logic.